libcob/cconv.c

```diff
@@ -341,21 +380,83 @@ const cob_u8_t cob_ascii_ibmebcdic[256] = {
 	0x3F, 0x3F, 0x3F, 0x3F, 0x3F, 0x3F, 0x3F, 0x3F
 };
 
-enum ebcdic_table cb_ebcdic_table = CB_EBCDIC_DEFAULT;
+const char *
+cob_get_collation_id_as_string (int col_id)
+{
+	switch (col_id) {
+	case CB_EBCDIC_DEFAULT: return "CB_EBCDIC_DEFAULT";
+	case CB_EBCDIC_RESTRICTED_GC: return "CB_EBCDIC_RESTRICTED_GC";
+	case CB_EBCDIC_IBM: return "CB_EBCDIC_IBM";
+	case CB_EBCDIC_GCOS: return "CB_EBCDIC_GCOS";
+	default: return NULL;
+	}
+}
 
-/* Decipher character conversion table names */
-int cobc_deciph_ebcdic_table_name (const char *name)
+int
+cob_get_collation_by_id (int col_id,
+			 const cob_u8_t **p_ebcdic_as_ascii,
+			 const cob_u8_t **p_ascii_as_ebcdic)
 {
-	if (! cb_strcasecmp (name, "DEFAULT")) {
-		cb_ebcdic_table = CB_EBCDIC_DEFAULT;
-	} else if (! cb_strcasecmp (name, "RESTRICTED-GC")) {
-		cb_ebcdic_table = CB_EBCDIC_RESTRICTED_GC;
-	} else if (! cb_strcasecmp (name, "IBM")) {
-		cb_ebcdic_table = CB_EBCDIC_IBM;
-	} else if (! cb_strcasecmp (name, "GCOS")) {
-		cb_ebcdic_table = CB_EBCDIC_GCOS;
-	} else {
-		return 1;
+	const cob_u8_t *ebcdic_as_ascii = NULL;
+	const cob_u8_t *ascii_as_ebcdic = NULL;
+
+	switch (col_id) {
+	case CB_EBCDIC_DEFAULT:
+		ebcdic_as_ascii = cob_ebcdic_ascii;
+		ascii_as_ebcdic = cob_ascii_ebcdic;
+		break;
+	case CB_EBCDIC_RESTRICTED_GC:
+		ebcdic_as_ascii = cob_ebcdic_ascii_alt;
+		ascii_as_ebcdic = cob_ascii_alt_ebcdic;
+		break;
+	case CB_EBCDIC_IBM:
+		ebcdic_as_ascii = cob_ibmebcdic_ascii;
+		ascii_as_ebcdic = cob_ascii_ibmebcdic;
+		break;
+	case CB_EBCDIC_GCOS:
+		ebcdic_as_ascii = cob_gcos7ebcdic_ascii;
+		ascii_as_ebcdic = cob_ascii_gcos7ebcdic;
+		break;
+	default:
+		return -1;
 	}
+
+	if (p_ebcdic_as_ascii != NULL) {
+		*p_ebcdic_as_ascii = ebcdic_as_ascii;
+	}
+
+	if (p_ascii_as_ebcdic != NULL) {
+		*p_ascii_as_ebcdic = ascii_as_ebcdic;
+	}
+
 	return 0;
 }
+
+int
+cob_get_collation_by_name (const char *col_name,
```

Comment:
especially as this is an API function: please add some function comments before it